### ISSUE TITLE
BCC: BattlefieldMap gets bricked when attempting to drag it

### ISSUE THREAD

**Issue by Ketho** [Stanzilla/WoWUIBugs#135]
When attempting to drag the Zone Map / BattlefieldMap it gives an error and appears in the middle of the screen, any mouse input also  gets eaten. After reloading and trying to show it again with `ctrl-m` it's setpointed at the bottomleft of the screen.

Video: https://user-images.githubusercontent.com/1073877/121725647-d6828200-cae9-11eb-9363-218767e8f44a.mp4
```
Message: ..\FrameXML\UIParent.lua line 2861:
   attempt to compare nil with number
Debug:
   [string "@..\FrameXML\UIParent.lua"]:2861: ValidateFramePosition()
   [string "@Blizzard_BattlefieldMap\Blizzard_BattlefieldMap.lua"]:71:
      ...\Blizzard_BattlefieldMap\Blizzard_BattlefieldMap.lua:44
```
When trying to drag it once it's at the bottomleft:
```
Message: ..\FrameXML\UIParent.lua line 2885:
   Action[SetPoint] failed because[SetPoint would result in anchor family connection]: attempted from: BattlefieldMapTab:SetPoint.
```

It can be reset by deleting the savedvariables `BattlefieldMapOptions = nil`
I'm not sure what's going wrong with the anchors or any supposed fix.

Build: 2.5.1 (38988)

**Comment by Meorawr** [comment]
The core is issue is that BattlefieldMapFrame is anchored [to both BattlefieldMapTab](https://github.com/Gethe/wow-ui-source/blob/861cef73e230aef353724092a8fe8dd1009fd4d6/AddOns/Blizzard_BattlefieldMap/Blizzard_BattlefieldMap.xml#L67-L70) and [to UIParent due to inheriting MapCanvasFrameTemplate](https://github.com/Gethe/wow-ui-source/blob/classic/AddOns/Blizzard_MapCanvas/Blizzard_MapCanvas.xml#L55).

When StartMoving is called on BattlefieldMapTab its own anchor points are erased and it's re-anchored to its current location on the screen - which causes a silent anchor restriction failure as BattlefieldMapFrame is now joined to both screen space (via the tab) and UIParent (via the MapCanvasTemplate) to occur immediately which invalidates the rect of the tab and leaves it and BattlefieldMapFrame uninteractive.

If the CENTER anchor to UIParent on BattlefieldMapFrame isn't present - such that it's only anchored to the tab - the issue doesn't occur.

**Comment by robeson1010** [comment]
Any solution on this bug?

**Comment by Meorawr** [comment]
Should be resolved in 2.5.1.39170 based off the diffs.
